qa327/frontend/sessions.py

```diff
@@ -201,26 +201,27 @@ class UpdateSession(LoggedInSession):
     def __init__(self, username):
         super().__init__(username)
 
-    def updateTicket(self, ticket_name, ticket_price, ticket_quantity):
-        helpers.TransactionsHelper.newTicketTransaction("update", self.username, ticket_name, ticket_price, ticket_quantity)
-        helpers.ResourcesHelper.getTicketInfo()[ticket_name]['price'] = ticket_price
-        helpers.ResourcesHelper.getTicketInfo()[ticket_name]['number'] = ticket_quantity
-
     def operate(self):
         try:
             ticket_name = helpers.UserIOHelper.acceptTicketName()
             ticket_quantity = helpers.UserIOHelper.acceptTicketQuantity()
             ticket_price = helpers.UserIOHelper.acceptTicketPrice()
-            date = helpers.UserIOHelper.acceptDate()
+            ticket_date = helpers.UserIOHelper.acceptDate()
             if ticket_name not in helpers.ResourcesHelper.getTicketInfo():
                 raise exceptions.WrongTicketNameException
-            self.updateTicket(self, ticket_name, ticket_price, ticket_quantity)        
+            self.updateTicket(ticket_name, ticket_price, ticket_quantity, ticket_date)        
         except exceptions.WrongFormatException as e:     
             print(str(e))
             print('\nUpdate failed, ending session...')
         except exceptions.WrongTicketNameException:
             print('\nThe ticket name you entered cannot be found, ending session...')
 
+    def updateTicket(self, ticket_name, ticket_price, ticket_quantity, ticket_date):
+        helpers.TransactionsHelper.newTicketTransaction("update", self.username, ticket_name, ticket_price, ticket_quantity, ticket_date)
+        helpers.ResourcesHelper.getTicketInfo()[ticket_name]['price'] = ticket_price
+        helpers.ResourcesHelper.getTicketInfo()[ticket_name]['number'] = ticket_quantity
+        helpers.ResourcesHelper.getTicketInfo()[ticket_name]['date'] = ticket_date
```

Comment:
why do you need these lines when you have passed everything in with line 220?

Reply:
> solved the conflict

Are you the smartest boy in the world? and the answer is YES!💥

Reply:
> why do you need these lines when you have passed everything in with line 220?

ASK ASK Cherish or Steven✊